From fbf90b850129b6507f64e4900dcef33f6981515a Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Fri, 6 May 2016 02:20:35 -0400
Subject: [PATCH] wr-policy: fix mailx send mail failure

* Port changes from Fedora 22:

  Reference sources: selinux-policy-3.13.1-128.21.fc22.src.rpm
  Changes come from: policy-f22-contrib.patch

  - allow postfix_postdrop_t connect to postfix_master_t socket
  - allow postfix_postdrop_t write to postfix_public_t socket file

* WRL fixes:

  - allow postfix_*_t search journal dir /run/systemd/journal

Fix avc denials:

  avc: denied { getattr write } for pid=871 comm="postdrop" \
  path="/var/spool/postfix/public/pickup" dev="hda" ino=27967 \
  scontext=root:sysadm_r:postfix_postdrop_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:postfix_public_t:s0 \
  tclass=sock_file permissive=0

  avc: denied { connectto } for pid=885 comm="postdrop" \
  path="/var/spool/postfix/public/pickup" \
  scontext=root:sysadm_r:postfix_postdrop_t:s0-s15:c0.c1023 \
  tcontext=system_u:system_r:postfix_master_t:s0-s15:c0.c1023 \
  tclass=unix_stream_socket permissive=1

  avc: denied { search } for pid=871 comm="postdrop" \
  name="journal" dev="tmpfs" ino=8217 \
  scontext=root:sysadm_r:postfix_postdrop_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 \
  tclass=dir permissive=0

  avc: denied { search } for pid=873 comm="cleanup" \
  name="journal" dev="tmpfs" ino=8217 \
  scontext=system_u:system_r:postfix_cleanup_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 \
  tclass=dir permissive=0

  avc: denied { search } for pid=875 comm="local" \
  name="journal" dev="tmpfs" ino=8217 \
  scontext=system_u:system_r:postfix_local_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 \
  tclass=dir permissive=0

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/services/postfix.te | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/policy/modules/services/postfix.te b/policy/modules/services/postfix.te
index ae434d0..30106ac 100644
--- a/policy/modules/services/postfix.te
+++ b/policy/modules/services/postfix.te
@@ -415,6 +415,8 @@ corenet_tcp_sendrecv_kismet_port(postfix_cleanup_t)
 mta_read_aliases(postfix_cleanup_t)
 mta_map_aliases(postfix_cleanup_t)
 
+mls_file_read_all_levels(postfix_cleanup_t)
+
 optional_policy(`
 	dkim_stream_connect(postfix_cleanup_t)
 ')
@@ -449,6 +451,8 @@ mta_map_aliases(postfix_local_t)
 mta_read_config(postfix_local_t)
 mta_send_mail(postfix_local_t)
 
+mls_file_read_all_levels(postfix_local_t)
+
 tunable_policy(`postfix_local_write_mail_spool',`
 	mta_manage_spool(postfix_local_t)
 ')
@@ -619,10 +623,12 @@ optional_policy(`
 allow postfix_postdrop_t self:capability sys_resource;
 
 rw_fifo_files_pattern(postfix_postdrop_t, postfix_public_t, postfix_public_t)
+rw_sock_files_pattern(postfix_postdrop_t, postfix_public_t, postfix_public_t)
 
 manage_files_pattern(postfix_postdrop_t, postfix_spool_maildrop_t, postfix_spool_maildrop_t)
 
 allow postfix_postdrop_t postfix_local_t:unix_stream_socket { read write };
+allow postfix_postdrop_t postfix_master_t:unix_stream_socket connectto;
 
 # for /var/spool/postfix/public/pickup
 stream_connect_pattern(postfix_postdrop_t, postfix_public_t, postfix_public_t, postfix_master_t)
@@ -635,6 +641,8 @@ term_dontaudit_use_all_ttys(postfix_postdrop_t)
 
 mta_rw_user_mail_stream_sockets(postfix_postdrop_t)
 
+mls_file_read_all_levels(postfix_postdrop_t)
+
 optional_policy(`
 	apache_dontaudit_rw_fifo_file(postfix_postdrop_t)
 ')
-- 
2.7.4

